lib/airrecord/table.rb

```diff
@@ -262,13 +249,14 @@ def client
     end
 
     def type_cast(value)
-      if value =~ /\d{4}-\d{2}-\d{2}/
-        Time.parse(value + " UTC")
-      else
-        value
-      end
+      return Time.parse(value + " UTC") if value =~ /\d{4}-\d{2}-\d{2}/
+      value
     end
 
+    def cast_association(value)
+      return value.map { |obj| obj&.id || obj } if value.respond_to?(:map)
+      [value&.id || value]
```

Comment:
```ruby
Array(value).map { |obj| obj&.id || obj }
```

Feels a little more natural to me here.

Reply:
We need a bit of test coverage here, e.g. no test fails if I change `[value&.id || value]` to `[value&.id]`.

Reply:
Nine years of writing Ruby, and I had no idea `Kernel.Array()` existed! That's so much nicer.

I ended up dropping this method and implementing it inline in the association setter. When you flagged that you could drop the conditional, I realized I had been conflating association setters and raw column setters here. Association setters always expect a table instance (or an array of table instances), so we can just call `Array(value).map(&:id)`.